WooCommerce/Classes/ViewRelated/Dashboard/Onboarding/StoreOnboardingView.swift

```diff
@@ -5,53 +5,69 @@ struct StoreOnboardingView: View {
     private let viewModel: StoreOnboardingViewModel
     private let taskTapped: (StoreOnboardingTask) -> Void
     private let viewAllTapped: (() -> Void)?
+    private let dismissAction: (() -> Void)?
 
     init(viewModel: StoreOnboardingViewModel,
          taskTapped: @escaping (StoreOnboardingTask) -> Void,
-         viewAllTapped: (() -> Void)? = nil) {
+         viewAllTapped: (() -> Void)? = nil,
+         dismissAction: (() -> Void)? = nil) {
         self.viewModel = viewModel
         self.taskTapped = taskTapped
         self.viewAllTapped = viewAllTapped
+        self.dismissAction = dismissAction
     }
 
     var body: some View {
-        VStack(alignment: viewModel.isExpanded ? .center : .leading, spacing: Layout.verticalSpacing) {
-            // Progress view
-            StoreSetupProgressView(isExpanded: viewModel.isExpanded,
-                                   totalNumberOfTasks: viewModel.taskViewModels.count,
-                                   numberOfTasksCompleted: viewModel.numberOfTasksCompleted)
-
-            // Task list
-            VStack(alignment: .leading, spacing: Layout.verticalSpacingBetweenTasks) {
-                ForEach(Array(viewModel.taskViewModels.enumerated()), id: \.offset) { index, taskViewModel in
-                    let isLastTask = index == viewModel.taskViewModels.count - 1
-
-                    StoreOnboardingTaskView(viewModel: taskViewModel,
-                                            showDivider: !isLastTask) { task in
-                        taskTapped(task)
+        VStack {
+            Color(.systemColor(.systemGray6))
+                .frame(height: 16)
+                .renderedIf(!viewModel.isExpanded)
+
+            let verticalSpacing = viewModel.isExpanded ? Layout.VerticalSpacing.expandedMode : Layout.VerticalSpacing.collapsedMode
+            VStack(alignment: viewModel.isExpanded ? .center : .leading, spacing: verticalSpacing) {
+                DismissButton(action: dismissAction)
+                    .renderedIf(viewModel.isExpanded)
+
+                // Progress view
+                StoreSetupProgressView(isExpanded: viewModel.isExpanded,
+                                       totalNumberOfTasks: viewModel.taskViewModels.count,
+                                       numberOfTasksCompleted: viewModel.numberOfTasksCompleted)
+
+                // Task list
+                VStack(alignment: .leading, spacing: Layout.verticalSpacingBetweenTasks) {
+                    ForEach(Array(viewModel.tasksForDisplay.enumerated()), id: \.offset) { index, taskViewModel in
+                        let isLastTask = index == viewModel.tasksForDisplay.count - 1
```

Comment:
💡 Since we don't need the index of each item, it's best to traverse through the task list and compare the current item with the last one in the list. Identifying objects with `offset` is not a recommended thing to do for performance.
```suggestion
                    ForEach(viewModel.tasksForDisplay) { taskViewModel in
                        let isLastTask = taskViewModel == viewModel.tasksForDisplay.last
```

(the `TaskViewModel` will need to conform to `Equatable` and `Identifier` in this case)

Reply:
Done in 65e14ee0aebe163518ab1b243cde532019ad8af8